### PR TITLE
Moe writing branch from 7f1714ba3d3681bed29d8238468533e63c07ba9f

### PR DESCRIPTION
**Pull request by sameb**
Merge internal changes.

### CODE REVIEW

File: extensions/servlet/src/com/google/inject/servlet/ServletScopes.java [hunk 1]
```diff
@@ -218,9 +220,15 @@ public String toString() {
    *     thread, or if the request has completed.
    * 
    * @since 3.0
+   * @deprecated You probably want to use {@code transferRequest} instead
```
Comment (tavianator) [inline extensions/servlet/src/com/google/inject/servlet/ServletScopes.java]: :(

We currently use this to provide access to `@RequestScoped` objects inside thread pools.  `transferRequest()` doesn't allow parallelism.  I suppose we could use `scopeRequest()` and manually forward a few things like the `HttpServletRequest`.  Is that the recommended way now?

Comment (sameb) [inline extensions/servlet/src/com/google/inject/servlet/ServletScopes.java]: We've found continueRequest to have a variety of issues.  Internally, some teams have taken to using a rewritten `@RequestScope` implementation that allows access through background threads (for types that are guaranteed to be immutable, or whitelisted types).

If continueRequest is working for you, no reason to stop using it... but be aware that it makes it easy to accidentally access mutable objects from many threads at once.

File: pom.xml [hunk 2]
```diff
@@ -159,7 +159,7 @@ See the Apache License Version 2.0 for the specific language governing permissio
       <dependency>
         <groupId>cglib</groupId>
         <artifactId>cglib</artifactId>
-        <version>3.1</version>
+        <version>3.2</version>
```
Comment (mcculls) [inline pom.xml]: FYI the version on central appears to be 3.2.0: http://search.maven.org/#artifactdetails%7Ccglib%7Ccglib%7C3.2.0%7Cjar

Comment (sameb) [inline pom.xml]: Thanks, Stuart.  I'll update this.

File: extensions/servlet/src/com/google/inject/servlet/ServletScopes.java (hunk 1 above)
Comment (tavianator) [inline extensions/servlet/src/com/google/inject/servlet/ServletScopes.java]: We do something similar, with a custom `@ThreadSafeRequestScoped` scope.  We put those objects in the `seedMap` to share them, and rely on `continueRequest()` to re-create new instances of the non-thread-safe `@RequestScoped` objects in the other threads.  I agree that `continueRequest()` on its own isn't perfect.

Well, as long as deprecated doesn't mean "getting deleted soon" I'll keep using it :).

**Comment by cgdecker** [comment]
LGTM assuming builds succeed now.

**Comment by mcculls** [comment]
Tested locally with various projects that use AOP and guice-servlet, looks good.

**Comment by sameb** [comment]
Thanks for checking Stuart!

**Pull request merged**